element_deeplabcut/model.py

```diff
@@ -603,41 +602,53 @@ def infer_output_dir(cls, key: dict, relative: bool = False, mkdir: bool = False
         return output_dir.relative_to(processed_dir) if relative else output_dir
 
     @classmethod
-    def insert_estimation_task(
+    def generate(
         cls,
-        key: dict,
-        task_mode: str = "trigger",
-        params: dict = None,
-        relative: bool = True,
-        mkdir: bool = True,
-        skip_duplicates: bool = False,
+        video_recording_key: dict,
+        model_name: str,
+        *,
+        task_mode: str = None,
+        analyze_videos_params: dict = None,
     ):
         """Insert PoseEstimationTask in inferred output dir.
 
         Based on the convention / video_dir / device_{}_recording_{}_model_{}
 
         Args:
-            key: DataJoint key specifying a pairing of VideoRecording and Model.
-            task_mode (bool): Default 'trigger' computation. Or 'load' existing results.
-            params (dict): Optional. Parameters passed to DLC's analyze_videos:
+            video_recording_key (dict): DataJoint key specifying a pairing of VideoRecording.
+            model_name (str): Name of DLC model (from Model table) to be used for inference.
+            task_mode (str): Default 'trigger' computation. Or 'load' existing results.
+            analyze_videos_params (dict): Optional. Parameters passed to DLC's analyze_videos:
                 videotype, gputouse, save_as_csv, batchsize, cropping, TFGPUinference,
                 dynamic, robust_nframes, allow_growth, use_shelve
-            relative (bool): Default True. Report directory relative to get_dlc_processed_data_dir().
-            mkdir (bool): Default True. Make directory if it doesn't exist.
-            skip_duplicates (bool): Default False. If true skips insertion of duplicate keys
         """
-        output_dir = cls.infer_output_dir(key, relative=relative, mkdir=mkdir)
+        processed_dir = get_dlc_processed_data_dir()
+        output_dir = cls.infer_output_dir(
+            video_recording_key, relative=False, mkdir=True
+        )
+
+        if task_mode is None:
+            try:
+                _ = dlc_reader.PoseEstimation(output_dir)
+            except FileNotFoundError:

```

Comment:
Are you sure this would hit `FileNotFound`? - I think we use assertions to ensure requisite files are available.
```suggestion
            except AssertionError:

```

Reply:
Right, I'll change those to `FileNotFound` for better error clarity

Reply:
fixed